### DISCUSSION TITLE
How do I calculate expected future lifetime?

### DISCUSSION THREAD

**Discussion by shivam017arora** [sebp/scikit-survival#222] [Q&A] [answered]
<img width="1107" alt="Screenshot 2021-09-16 at 7 55 38 PM" src="https://user-images.githubusercontent.com/26146104/133629956-208d0ec6-2f99-4308-a15a-c3b874d16ad0.png">

I am trying to do customer data analysis and trying to predict the expected future lifetime of the customers. I am using Cox Model and Random Forest Model. 

Please help as I am unable to find any support for this in the library. Is there a way I can calculate this?

**Comment by sebp** [comment, marked as answer]
You would need to integrate over the survival function as in the text you posted (e.g. using [trapezoid rule](https://docs.scipy.org/doc/scipy-1.7.0/reference/generated/scipy.integrate.trapezoid.html#scipy.integrate.trapezoid)). The problem is the infinity part in the formula, which makes it difficult if the survival function does not reach zero at some observed time point. There are ways around this by making additional assumptions, as I mentioned in [this comment](https://github.com/sebp/scikit-survival/issues/190#issuecomment-806444984-permalink).

PR #196 started to implement this, but unfortunately the person never finished it.

**Reply by shivam017arora** [reply]
![image](https://user-images.githubusercontent.com/26146104/134166559-77783a97-788d-40b3-a314-a783b076cfa4.png)


Am I doing this right?

**Reply by sebp** [reply]
That looks correct. But it will only really estimate the expectation **if** the probability of the last time point is zero (`X.y[-1] == 0`), as I [explained previously](https://github.com/sebp/scikit-survival/issues/190#issuecomment-806444984-permalink).

**Reply by shivam017arora** [reply]
Thank you so much for your time and kindness! Appreciate it!
